packages/core/src/checkout-buttons/strategies/braintree/braintree-paypal-credit-button-strategy.ts

```diff
@@ -0,0 +1,203 @@
+        shippingAddress?: Address | null,
+        onError?: (error: BraintreeError | StandardError) => void
+    ): Promise<string> {
+        const state = await this._store.dispatch(this._checkoutActionCreator.loadDefaultCheckout());
+
+        try {
+            const checkout = state.checkout.getCheckoutOrThrow();
+            const config = state.config.getStoreConfigOrThrow();
+            const customer = state.customer.getCustomer();
+
+            const address = shippingAddress || customer?.addresses?.[0];
+            const shippingAddressOverride = address ? mapToBraintreeShippingAddressOverride(address) : undefined;
+
+            return await braintreePaypalCheckout.createPayment({
+                flow: 'checkout',
+                enableShippingAddress: true,
+                shippingAddressEditable: false,
+                shippingAddressOverride,
+                amount: checkout.outstandingBalance,
+                currency: config.currency.code,
+                offerCredit: true,
+            });
+        } catch (error) {
+            if (onError) {
+                onError(error);
+            }
+
+            throw error;
+        }
+    }
+
+    private async _tokenizePayment(
+        authorizeData: PaypalAuthorizeData,
+        braintreePaypalCheckout: BraintreePaypalCheckout,
+        methodId: string,
+        shouldProcessPayment?: boolean,
+        onError?: (error: BraintreeError | StandardError) => void
+    ): Promise<BraintreeTokenizePayload> {
+        try {
+            const { deviceData } = await this._braintreeSDKCreator.getDataCollector({ paypal: true });
+            const tokenizePayload = await braintreePaypalCheckout.tokenizePayment(authorizeData);
+            const { details, nonce } = tokenizePayload;
+
+            this._formPoster.postForm('/checkout.php', {
+                payment_type: 'paypal',
+                provider: methodId,
+                action: shouldProcessPayment ? 'process_payment' : 'set_external_checkout',
+                nonce,
+                device_data: deviceData,
+                billing_address: JSON.stringify(mapToLegacyBillingAddress(details)),
+                shipping_address: JSON.stringify(mapToLegacyShippingAddress(details)),
+            });
+
+            return tokenizePayload;
+        } catch (error) {
+            if (onError) {
+                onError(error);
+            }
+
+            throw error;
+        }
+    }
+
+    private _handleError(
+        error: BraintreeError,
+        containerId: string,
+        onErrorCallback?: (error: BraintreeError) => void
+    ): void {
+        this._hideElement(containerId);
+
+        if (onErrorCallback) {
+            onErrorCallback(error);
+        }
+    }
+
+    private _hideElement(elementId?: string): void {
+        const element = elementId && document.getElementById(elementId);
+
+        if (element) {
+            Object.assign(element.style, { display: 'none' });
```

Comment:
We should remove the element, better to clean up the elements if something doesn't go right.